### DISCUSSION TITLE
Goto type definition difference when goto on type

### DISCUSSION THREAD

**Discussion by ckipp01** [scalameta/metals#4978] [General]
I came across this when looking into https://github.com/lampepfl/dotty/discussions/16858 and I thought it was interested. I'm not 100% sure what should happen here. But you given a piece of code like this:

```scala
val list: List[Int] = List(1, 2, 3)
//    ^1    ^2
```

If you trigger a goto definition on position `2` it will lead you to `package.scala` at:

```scala
  type List[+A] = scala.collection.immutable.List[A]
  val List = scala.collection.immutable.List
```

Whereas if you trigger a goto type definition at position `1` you're then lead to `List.scala`

```scala
sealed abstract class List[+A]
```

I sort of assumed that they behavior on the goto definition of `List` and the goto type definition here to be the same. What do others think here?

**Comment by tgodzik** [comment]
This seems like https://contributors.scala-lang.org/t/transparent-term-aliases/5553/13 :fearful: 

For go to definition we don't dealias here, and when we check the type for `go to type definition` we get the correct list.
